odd-platform-ui/src/lib/hooks/api/terms.ts

```diff
@@ -29,3 +35,45 @@ export function useGetTermLinkedColumns(params: TermApiGetTermLinkedColumnsReque
     initialData: { pageInfo: { total: 0, hasNext: false }, items: [] },
   });
 }
+
+export function useAssignTermQueryExample(termId: number) {
+  const queryClient = useQueryClient();
+
+  return useMutation({
+    mutationKey: ['assignTermQueryExample', termId],
+    mutationFn: async ({
+      queryExampleTermFormData,
+    }: TermApiCreateQueryExampleToTermRelationshipRequest) =>
+      termApi.createQueryExampleToTermRelationship({
+        queryExampleTermFormData,
+        termId,
+      }),
+    onSuccess: async () => {
+      showSuccessToast({ message: 'Query Example successfully assigned!' });
+      await queryClient.invalidateQueries({
+        queryKey: ['getQueryExamplesByTermId'],
+      });
+    },
+  });
+}
+
+export function useUnassignTermQueryExample(termId: number) {
+  const queryClient = useQueryClient();
+
+  return useMutation({
+    mutationKey: ['unassignTermQueryExample'],
+    mutationFn: async ({
+      exampleId,
+    }: TermApiDeleteQueryExampleToTermRelationshipRequest) =>
+      termApi.deleteQueryExampleToTermRelationship({
+        termId,
+        exampleId,
+      }),
+    onSuccess: async () => {
+      showSuccessToast({ message: 'Query Example successfully unassigned!' });
+      await queryClient.invalidateQueries({
+        queryKey: ['getQueryExamplesByTermId'],
+      });
+    },
+  });
+}
```

Comment:
We don't need to use arg for termId at this level. So mutation func with two args is more obvious behaviour. Same for the `useAssignTermQueryExample` hook
```suggestion
export function useUnassignTermQueryExample() {
  const queryClient = useQueryClient();

  return useMutation({
    mutationKey: ['unassignTermQueryExample'],
    mutationFn: async ({
      exampleId,
      termId
    }: TermApiDeleteQueryExampleToTermRelationshipRequest) =>
      termApi.deleteQueryExampleToTermRelationship({
        termId,
        exampleId,
      }),
    onSuccess: async () => {
      showSuccessToast({ message: 'Query Example successfully unassigned!' });
      await queryClient.invalidateQueries({
        queryKey: ['getQueryExamplesByTermId'],
      });
    },
  });
}
```

Reply:
done